src/inspector/network_resource_manager.h

```diff
@@ -0,0 +1,36 @@
+// network_resource_manager.h
+#ifndef SRC_INSPECTOR_NETWORK_RESOURCE_MANAGER_H_
+#define SRC_INSPECTOR_NETWORK_RESOURCE_MANAGER_H_
+
+#include <atomic>
+#include <string>
+#include <unordered_map>
+#include "node_mutex.h"
+
+namespace node {
+namespace inspector {
+
+class NetworkResourceManager {
+ public:
+  void Put(const std::string& url, const std::string& data);
+  std::string Get(const std::string& url);
+
+  // Accessor to get URL for a given stream id
+  std::string GetUrlForStreamId(uint64_t stream_id);
+  // Erase resource and mapping by stream id
+  void EraseByStreamId(uint64_t stream_id);
+  // Returns the stream id for a given url, or 0 if not found
+  uint64_t GetStreamId(const std::string& url);
+
+ private:
+  uint64_t NextStreamId();
+  std::unordered_map<std::string, std::string> resources_;
```

Comment:
Alternatively, we can use the url as id as well, so that we don't have to index the resources with an extra int id.

The protocol defines that `type StreamHandle extends string`, so a url can be `StreamHandle`.